File: src/main/java/application/history/history/application/services/UpdateHistoryContentService.java
package application.history.history.application.services;

import application.history.history.application.inputs.UpdateHistoryInput;
import application.history.history.domain.entities.History;
import application.history.history.domain.repositories.HistoryRepository;
import org.springframework.stereotype.Service;

import java.util.Date;

@Service
public class UpdateHistoryContentService {
    private final HistoryRepository repository;

    public UpdateHistoryContentService(HistoryRepository repository) {
        this.repository = repository;
    }

    public History run(Date date, UpdateHistoryInput request) {
        History history = this.repository.findByDate(date).orElseThrow();

        history.updateContent(request.getContent());
        return this.repository.save(history);
    }
}

File: src/main/kotlin/com/example/history/application/services/UpdateHistoryContentService.kt
package com.example.history.application.services

import com.example.history.application.inputs.UpdateHistoryInput
import com.example.history.domain.entities.History
import com.example.history.domain.repositories.HistoryRepository
import org.springframework.stereotype.Service
import java.time.LocalDate

@Service
class UpdateHistoryContentService(repository: HistoryRepository) {
    private val repository: HistoryRepository

    fun run(date: LocalDate, request: UpdateHistoryInput): History {
        val history: History = repository.findByDate(date).orElseThrow()
        history.updateContent(request.getContent())
        return repository.save(history)
    }

    init {
        this.repository = repository
    }
}
